api/src/main/java/org/apache/iceberg/catalog/SupportsNamespaces.java

```diff
@@ -78,7 +78,7 @@ default List<Namespace> listNamespaces() {
    * List namespaces from the namespace.
    *
    * <p>For example, if table a.b.t exists, use 'SELECT NAMESPACE IN a' this method must return
-   * Namepace.of("a","b") {@link Namespace}.
+   * Namespace.of("a","b") {@link Namespace}.
    *
    * @return a List of namespace {@link Namespace} names
    * @throws NoSuchNamespaceException If the namespace does not exist (optional)
```

Comment:
Does anyone know whether this API should return just next-level namespaces or all the children? Because I can see some inconsistency in the catalog implementation. I think we should enforce/mention it clearly here.

cc: @rdblue, @RussellSpitzer, @Fokko

Reply:
which catalogs are inconsistent at this moment?

Reply:
**Have not spent much time on this. But it looks like.** 

- Hadoop gives all namespaces, Nessie too (being fixed), Dynamodb (code looks like there is no level check. But I need to test it out. I am not sure yet), 

- glue & hive may be not applicable as it doesn't support nested namespaces.

- ECS, JDBC is giving only next-level namespaces.

- REST API spec has an example with just a 2-level namespace. So, cannot deduce there also. But from the description maybe it also gives next-level ("certain level")

I can check more on this tomorrow.

Reply:
Thanks for the investigation! @nastra do you know if there is a defined behavior for REST? If not I think we should define that at API level.

Reply:
For REST it should only return the child elements of the Namespace. So if you have two tables `a.b.c.table` & `a.b.d.table` and list namespaces:
* with `Namespace.empty()` -> should return `Namespace.of("a")`
* with `Namespace.of("a")` -> should return `Namespace.of("a.b")`
* with `Namespace.of("a.b")` -> should return `Namespace.of("a.b.c")` & `Namespace.of("a.b.d")`
* with `Namespace.of("a.b.c")` -> should return nothing, because there are no child namespaces

I think it would be good to make the Javadoc clearer with an example like the above. Additionally, we should probably update `CatalogTests.testListNestedNamespaces()` so show this example.

Reply:
I also think the assumption was we list only direct children. Otherwise, we would need to support nested namespaces in places like `listTables()`.

Reply:
I also think it should be documented in the API and all implementations must conform (which means fixing others?).

Reply:
I checked Spark catalog APIs and it also assumes we only list direct children.

Reply:
Thanks for checking. I have Updated the javadoc now. Nessie side listNamespace bug was also fixed in another PR recently. 
This PR I would like to keep it for javadoc changes, in a follow up I can add a testcases for all the catalogs and fix the code if any catalog has issue.